roles/setup_pem/tasks/main.yml

```diff
@@ -43,16 +43,30 @@
     - pem_server_exists
   with_dict: "{{ servers }}"
 
+- name: Create pemagent user on primary
+  import_tasks: pem_agent_user.yml
+  when: pem_server_exists
+  delegate_to: "{{ primary_public_ip }}"
+
 - name: Prepare hba host ip lists
   include_tasks: prepare_hba_value_list.yml
   when: 
     - item.value.node_type in ['primary','standby']
     - pem_server_exists
   with_dict: "{{ servers }}"
 
+- name: Configure pemserver
+  include_tasks: setup_pemserver.yml
+  when: 
+     - item.value.node_type == 'pemserver'
+     - pem_server_exists
+  with_dict: "{{ servers }}"
+
 - name: Execute setup_pem for all nodes
-  include_tasks: setup_pem.yml
-  when: pem_server_exists
+  include_tasks: setup_pemagent.yml
+  when: 
+     - item.value.node_type != 'pemserver'
```

Comment:
No need to setup a PEM agent on PEM server database ?

Reply:
Yes. setup_pemserver automatically installs pemagent for pemserver. Also, pemagent on pemserver is different from the pemagent on other nodes like primary and standby